homeassistant/components/viam/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""The viam integration."""
+
+from __future__ import annotations
+
+from viam.app.viam_client import ViamClient
+from viam.rpc.dial import Credentials, DialOptions
+
+from homeassistant.const import CONF_ADDRESS, CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_API_ID,
+    CONF_CREDENTIAL_TYPE,
+    CONF_SECRET,
+    CRED_TYPE_API_KEY,
+    DOMAIN,
+)
+from .manager import ViamConfigEntry, ViamManager
+from .services import async_setup_services
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ViamConfigEntry) -> bool:
```

Comment:
```suggestion
async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Reply:
So we can't pass the config entry through, we should fetch it during the service call. Checkout mealie